### PR TITLE
Refactor (frontend): mixpanel as hook

### PR DESCRIPTION
**Pull request by luixo**
A little bit of hooks :)

### CODE REVIEW

File: frontend/src/hooks/analytics/use-analytics-track-on-mount.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { Dict } from "mixpanel-browser";
+import { useEffect } from "react";
+import { useAnalyticsTrack } from "./use-analytics-track";
+
+export const useAnalyticsTrackOnMount = (event: string, args?: Dict) => {
```
Comment (shelegdmitriy) [inline frontend/src/hooks/analytics/use-analytics-track-on-mount.ts:5]: If I'm not mistaken we have to use `useAnalyticsTrackOnMount` on pages only? So, maybe it's better to provide an explanation above?

File: frontend/src/hooks/analytics/use-analytics-track.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { useCallback } from "react";
+import mixpanel, { Dict } from "mixpanel-browser";
+
+export const useAnalyticsTrack = () => {
```
Comment (shelegdmitriy) [inline frontend/src/hooks/analytics/use-analytics-track.ts:4]: `useAnalyticsTrack` we have to use in components only?

**Review by shelegdmitriy — APPROVED** [review]

File: frontend/src/hooks/analytics/use-analytics-track-on-mount.ts (hunk 1 above)
Comment (luixo) [reply frontend/src/hooks/analytics/use-analytics-track-on-mount.ts:5]: Well it depends on our goals, we may want to log not the page mount, but specific component.
At the moment it is true, but one can easily find it out by looking up hook usage.

File: frontend/src/hooks/analytics/use-analytics-track.ts (hunk 2 above)
Comment (luixo) [reply frontend/src/hooks/analytics/use-analytics-track.ts:4]: I didn't get the question :)
We can use it everywhere where we want track something to analytics on action.

**Pull request merged**